### PR TITLE
Mx4PC Operator 2.17.0 release notes (planned for release around June 5)

### PR DESCRIPTION
**Pull request by zlogic**
We have a major feature release planned somewhere in the middle (or end) of next week, and this time there's a bit more documentation than usual. To provide a bit more flexibility and have an early start, this is the draft documentation explaining how to use Azure Managed Identities.

We'll provide complete release notes a bit later in this PR.

@Nidhi251289 could you please do a review from your side?

### CODE REVIEW

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md [hunk 1]
```diff
@@ -215,6 +220,11 @@ Azure PostgreSQL databases require additional firewall configuration to allow co
 
 Some managed PostgreSQL databases might have restrictions or require additional configuration.
 
+As an alternative to static password authentication, Mendix Operator can use its Kubernetes Service Account to authenticate with:
+
+* AWS RDS databases using IAM roles
+* Azure Database for PostgreSQL (Flexible Server) databases using workload identities
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: Instead of workload identity, should we not use Managed identity

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md [hunk 2]
```diff
@@ -254,6 +264,8 @@ Amazon and Azure SQL servers require additional firewall configuration to allow
 
 Some managed SQL Server databases might have restrictions or require additional configuration.
 
+As an alternative to static password authentication, Mendix Operator can use its Kubernetes Service Account to authenticate with Azure SQL databases.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: can we also add in the end of sentence

using managed identities

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md [hunk 3]
```diff
@@ -319,9 +331,10 @@ configuration details.
 
 ### 5.4 Azure Blob Storage
 
-An existing [Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/) container can be attached to Mendix app environments.
+[Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/) is supported.
+Mendix Operator can either provide a static access key and other credentials directly to environments, or manage containers and provide access to environments through Azure Managed Identities.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: this line looks odd and confusing to me. can we rephrase it to:

Mendix Operator can provide credentials to the environments either using static access key or using Azure Managed Identities

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md [hunk 4]
```diff
@@ -319,9 +331,10 @@ configuration details.
...
 
-Unlike MinIO and S3, Mendix for Private Cloud doesn't manage Azure Blob Storage containers or accounts.
+A complete list of supported Azure Blob Storage modes and their required IAM permissions for each one is available in [storage plan](/developerportal/deploy/standard-operator/#storage-plan) configuration details.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: is it not that "IAM permissions" is more associated to AWS. Should we rather use **role assignments** here

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md (hunk 2 above)
Comment (Nidhi251289) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: Should we also add the Mendix version supported for Managed identities

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 5]
```diff
@@ -86,6 +88,16 @@ If your app is created in Mendix 9.22 or above, and its Kubernetes service accou
 For more information and a complete walkthrough example, see the [AWS RDS IAM authentication example](#configure-using-rds-iam).
 {{% /alert %}}
 
+{{% alert color="info" %}}
+If your app is created in Mendix 9.22 or above, and its Kubernetes service account is linked to an Azure Managed Identity, you do not need to specify a `database-password` to access an Azure Postgres database. Instead, you can use the same managed identity for database authentication.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:91]: Does it mean that the azure managed identity also works with Mx 9.22 version. I understood that it will work from 10.10 version onwards.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 6]
```diff
@@ -694,6 +706,94 @@ To enable your environment to use [Azure Key Vault](https://learn.microsoft.com/
 
 For more information, see the [Azure Key Vault Provider](https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/usage/#create-your-own-secretproviderclass-object) example in the Azure Key Vault documentation.
 
+#### 3.3.1 Using Managed Identity authentication for Postgres databases {#configure-using-azwi-postgres}
+
+[Azure Postgres (Flexible Server)](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/service-overview) databases can use [workload identity authentication](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-connect-with-managed-identity) instead of regular passwords.
+
+{{% alert color="warning" %}}Azure has a legacy Postgres (Single Server) database. The Mendix Operator doesn't support workload [identity authentication for Single Server databases](https://learn.microsoft.com/en-us/azure/postgresql/single-server/how-to-configure-sign-in-azure-ad-authentication), only the new Flexible Server is supported. This section only applies to Flexible Server databases.{{% /alert %}}
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md]: Change workload to managed identity

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 7]
```diff
@@ -694,6 +706,94 @@ To enable your environment to use [Azure Key Vault](https://learn.microsoft.com/
...
+1. Remove or comment out `database-password` from the `SecretProviderClass` and the associated Key vault Secret.
+2. Enable [Microsoft Entra authentication](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-configure-sign-in-azure-ad-authentication) and add yourself as an Entra Admin user in the Postgres database.
+3. Open Azure Cloud Shell (or a Bash-compatible terminal) and run `az login` to authenticate with Entra ID.
+4. Run the following command to connect to the Azure Postgres database, replacing `<hostname>` with the Postgres server hostname (e.g. `example.postgres.database.azure.com`), and `<user-email>` with your Entra ID login email (e.g. `user@tenant.onmicrosoft.com`):
+
+   ```shell {linenos=false}
+   psql "host=<hostname> port=5432 dbname=postgres user=<user-email> password=$(az account get-access-token --resource-type oss-rdbms --output tsv --query accessToken) sslmode=require"
+   ```
+
+5. In the Postgres client, run the following commands (replace `<database-user>` with the environment's current `database-user` name, and `<workload-identity-uuid>` with the **Object (principal) ID** of the environment's **Managed Identity**:
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md]: Before this step, do we have to manually create a managed identity

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md [hunk 8]
```diff
@@ -694,6 +706,94 @@ To enable your environment to use [Azure Key Vault](https://learn.microsoft.com/
...
+1. Remove or comment out `storage-azure-account-key` and `storage-azure-account-name` from the `SecretProviderClass` and the associated Key vault Secret.
+2. Add the following keys to the `SecretProviderClass` and Azure Key vault:
+    * `storage-azure-blob-endpoint` - specify the storage account's [Primary endpoint](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#standard-endpoints);
+    * `storage-azure-use-default-azure-credential` - set to `true` to enable managed identity authentication.
+3. Open the envionment's Blob Storage Account Container in the Azure portal, and [assign a Storage Blob Data Contributor role](https://learn.microsoft.com/en-us/azure/storage/blobs/assign-azure-role-data-access) to the environment's managed identity.
+4. Restart the Mendix app environment.
+
 ## 4 Additional Considerations {#additional-considerations}
 
 When implementing a secret store, keep in mind the following considerations:
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:798]: cna you also add below sentence:

1. You need to create as many federated credentials per environment when using secret key store in Azure.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 9]
```diff
@@ -174,12 +180,14 @@ The following database types are supported:
 
 Postgres databases can be used with [static authentication](#database-postgres-static).
 If the Postgres instance is an AWS RDS database, you can use [IAM authentication](#database-postgres-iam) for additional security.
+If the Postgres instance is an Azure Postgres (Flexible Server) database, you can use [managed identity authentication](#database-postgres-azwi) for additional security.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:188]: can u aso mention the operator version this will be supporting from

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 10]
```diff
@@ -371,6 +379,105 @@ AWS IRSA allows a Kubernetes Service Account to assume an IAM role. For this to
...
+
+##### 2.3.3.4 Create Workflow
+
+When a new environment is created, the Mendix Operator performs the following actions:
+
+* Create a Managed Identity for an environment. This Managed Identity will be created in the same resource group, subscription and region as the *Postgres Admin* managed identity.
+* Create a Kubernetes Service Account and attach it to the environment's Managed Identity. This Service Account acts as a replacement for static credentials, and can also be used to authenticate with Azure Postgres databases.
+* Generate a database name and username (Postgres role) for the new environment.
+* Create a new database in the provided Postgres server. This will be the environment's dedicated database.
+* Create a new user (Postgres role) for the new environment, and allow this user to access only the new environment's database. This will be the environment's user.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:427]: we already mentioned in line above that we create a new user.. why we are doing that again

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 11]
```diff
@@ -371,6 +379,105 @@ AWS IRSA allows a Kubernetes Service Account to assume an IAM role. For this to
...
+    * `mendix-operator`
+    * `mendix-agent`
+    * `mendix-storage-provisioner`
+  {{% /alert %}}
+
+{{% alert color="info" %}}
+To connect to an Azure Postgres database, the firewall must be configured to allow connections to the database from the Kubernetes cluster.
+{{% /alert %}}
+
+Azure workload identities allow a Kubernetes Service Account to authenticate itself as a specific Managed Identity. For this to work correctly, add a Federated Credential to the *Postgres Admin* managed identity:
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:469]: change workload to managed identities

Comment (Nidhi251289) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:469]: should we also mentioned that same federated credentials can be used for multiple environment. With this it also means that dedicated managed identity will only be created once.. lets say for postgres and not Azure blob storage.
Also, same service account can be used for multiple storage plans.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 12]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
 
 ### 3.4 Azure Blob Storage {#blob-azure}
 
+If you would like to have Mendix Operator with automation, and have full isolation between environments, use the [Azure managed identity authentication](#blob-azure-azwi) option. This option works with apps using Mendix 10.10 (or a later version).
+
+If you would like to simply share a container between environments, or to manually create a container and account per environment, use the [static credentials](#blob-azure-static) option.
+
+#### 3.4.1 Azure Blob Storage (Azure managed identity authentication){#blob-azure-azwi}
+
+This automated, on-demand option allows to use an existing blob storage accounts in multiple environments, and isolates environments from accessing each other's data.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1631]: I have a question: 

In regular authentication, we ask customer to provide container name, but in wi, we dont. Is there any specific reason for the same. I mean we can still provide an option to provide the Container name.
Also, we dont have On-demand and Dedicated option for Azure blob storage. While using managed identity, customer would still want to use a dedicated type for the storage plan.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 13]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
...
+* The storage account is shared.
+* Environment has full access only to its own container, cannot access data from other environments.
+
+##### 3.4.1.4 Create Workflow
+
+When a new environment is created, the Mendix Operator performs the following actions:
+
+* Create a Managed Identity for an environment. This Managed Identity will be created in the same resource group, subscription and region as the *Blob Storage Admin* managed identity.
+* Create a Kubernetes Service Account and attach it to the environment's Managed Identity. This Service Account acts as a replacement for static credentials, and can also be used to authenticate with the environment's Blob Storage Container.
+* Create a new container in the shared blob storage account. This will be the environment's dedicated container.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1663]: As i mentioned above, is it not possible to use an existing container to store images

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 14]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
...
+* Add the [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role to an envrionment's Managed Identity, scoped to its container.
+* Create a Kubernetes secret to provide connection details to the new app environment - to automatically configure the new environment.
+   Since the app environment will authenticate through a managed identity role, this secret will not contain any static passwords - only the database hostname, username and other non-sensitive connection details.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: When i checked the secret for SQL, i see   **MSSQL_SERVER_USER**: field as empty

`data:
  MSSQL_SERVER_HOST: bmlkaGktZGVzLXNxbC5kYXRhYmFzZS53aW5kb3dzLm5ldA==
  MSSQL_SERVER_PASSWORD: ""
  MSSQL_SERVER_PORT: MTQzMw==
  MSSQL_SERVER_USER: ""
  STRICT_TLS: dHJ1ZQ==`

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 15]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
...
+
+##### 3.4.1.5 Delete Workflow
+
+When an existing environment is deleted, the Mendix Operator performs the following actions:
+
+* Delete that environment's role assignment.
+* Delete that environment's container and its files.
+* Delete that environment's Managed Identity.
+* Delete that environment's Kubernetes Service Account.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1675]: If same service account is being used across multiple plans, then in that case, the service account should not be deleted.. Correct?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 16]
```diff
@@ -437,25 +551,121 @@ When a new environment is created, the Mendix Operator performs the following ac
...
+* Create a new contained database user for the new environment. This will be the environment's user, which only exists in the environment's database. This user will be [linked with the environment's Managed Identity](https://learn.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-configure?view=azuresql&tabs=azure-powershell#create-contained-users-mapped-to-microsoft-entra-identities).
+* Create a Kubernetes secret to provide connection details to the new app environment - to automatically configure the new environment.
+
+##### 2.5.2.5 Delete Workflow
+
+When an existing environment is deleted, the Mendix Operator performs the following actions:
+
+* Delete that environment's database (which also deletes the contained database user).
+* Delete that environment's Managed Identity.
+* Delete that environment's Kubernetes Service Account.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:656]: If same service account is being used across multiple plans, then in that case, the service account should not be deleted.. Correct?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 17]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
...
+* Create a new container in the shared blob storage account. This will be the environment's dedicated container.
+* Add the [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role to an envrionment's Managed Identity, scoped to its container.
+* Create a Kubernetes secret to provide connection details to the new app environment - to automatically configure the new environment.
+   Since the app environment will authenticate through a managed identity role, this secret will not contain any static passwords - only the database hostname, username and other non-sensitive connection details.
+
+##### 3.4.1.5 Delete Workflow
+
+When an existing environment is deleted, the Mendix Operator performs the following actions:
+
+* Delete that environment's role assignment.
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1672]: Any specific reason why we dont delete the role assigment for postgres database or do we not have any role assignments for Postgres db

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 18]
```diff
@@ -1365,42 +1609,141 @@ In the Amazon S3 plan configuration, enter the following details:
...
 * Delete that environment's Kubernetes blob file storage credentials secret.
 
-#### 3.4.6 Configuring the Plan
+##### 3.4.2.6 Configuring the Plan
 
 In the Azure Blob plan configuration, enter the following details:
 
-* **Account Name** and **Account Key** - Credentials for the blob storage container.
+* **Account Name** - Blob Storage account name.
+* **Managed Identity authentication** - Set to **yes**
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: if i undertsand it correctly, this section comes under static credentials for Azure blob storage, so should we not have below fields:

Account name
Managed Identity authentication set to No
Account key
Container name
Types

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md [hunk 19]
```diff
@@ -1782,3 +2125,158 @@ To create the required storage plans, do the following steps:
...
+   * Name of the storage account
+   * **Resource group**
+   * **Subscription ID**
+
+2. Grant the *storage admin* Managed Identity (created [in the beginning of this walkthrough](#walkthrough-azure-azwi)) the following permissions:
+
+  * A [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role scoped to the blob storage account.
+  * A [Role Based Access Control Administrator](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/general#role-based-access-control-administrator) role scoped to the blob storage account, and constrained to only have permissions to add the [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-contributor) role to Service principals.
+
+{{% alert color="info" %}}To improve security, set the firewall to only allow connections to the database from the Kubernetes cluster.{{% /alert %}}
```
Comment (Nidhi251289) [inline content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: Should we replace database with storage in sentence like:

To improve security, set the firewall to only allow connections to the storage from the Kubernetes cluster.{

**Review by Nidhi251289 — COMMENTED** [review]
Provided the comments

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 5 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md:91]: With Mendix 9.22, only Postgres supports managed identity auth (but not SQL Server or Azure Blob Storage).
With 10.10, all is good, everything works with a Managed Identity.
Technically, an older Mendix version (like 9.12) should also work with Postgres, but it would be unable to update its database password once the short-lived password expires. The app will still run, just will have to use only existing connections from its connection pool.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 7 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md]: Yes, but the Managed Identity should have already been created to grant access to Key Vault. The environment's Managed Identity grants access to everything - Key vault, database and blob storage. Or any other Azure service like AI/ChatGPT :)

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 9 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:188]: I mentioned it in the Limitations section (didn't want to overload the introduction)

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 10 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:427]: That was copy-pasted from other Postgres provisioners - I think it meant that first the database name (and password in 2.3.1.4) is generated, then the generated database name+username+password are used later in the process.

Would it make to update all Postgres provisioners and simplify them to something like "create a new database with a name generated based on the environment metadata"?

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 11 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:469]: I think Microsoft's documentation is a bit confusing:

1. Managed identity is like a user that can be granted access to specific resources, or who can join a group.
2. There are many ways something can authenticate itself as a Managed Identity, and Workload Identities is Microsoft's term for assigning a Managed Identity to a Kubernetes pod or another application.

> In Microsoft Entra, workload identities are applications, service principals, and managed identities.

When linking to a Microsoft document, I was using the same titles as mentioned in the documents (ideally redirect customers to the official Azure docs for any questions).

Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:469]: For the second comment, I mentioned it above in
> Mendix recommends using the same managed identity to manage Azure databases and blob storage containers, as this would be easier to set up and maintain

(to match the AWS IAM documentation)

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 12 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1631]: Indeed, with Managed Identities the idea is to keep apps and their data isolated. Only the Mendix Operator has permissions to create new containers, and when it creates a new environment, it generates a container name and grants the environment's Managed Identity permissions to only access that specific container.

With the previous plan, all apps were:
1. Running with full access to the entire storage account
2. If a container name is specified, apps would save all their files into one container, into its root directory. All files are in the same place, so it's difficult to understand which files belong to which app.

Dedicated plans existed to at least ensure that two apps cannot share the same container and credentials at the same time, but with the new automation it's no longer needed (every app is now isolated from others by default).

Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1631]: If there are enough requests, perhaps we could implement support for dedicated plans in a future release.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 13 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1663]: Indeed, if an existing container is used, all apps would be writing into the same directory and have access to each other's files, with the new option the Operator will automatically create per-environment containers.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 15 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1675]: If the environment is deleted, its Service Account and Managed Identity would also be deleted - to prevent a new environment from adopting this Service Account and its associated MI.

But the *storage admin* account (used by Mx4PC to create/delete managed identities) is not affected, only the envrionment's account would be deleted.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 16 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:656]: Same as above - if an environment is deleted, both of its storage plans would be deleted, and that also deletes the Managed Identity and Kubernetes Service Account. This would be done twice (by the database and file plans), one of them would perform the actual deletion, the other one would get an "already deleted" error and ignore it.

This is different from AWS, where the Service Account and IAM role are set up by the S3 provisioner, and the Postgres provisioner relies on S3 to set up permissions.

Azure has better conflict resolution, while AWS allows things to be done in any order.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 17 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md:1672]: Blob storage, key vault and many other services fully support Entra roles and rely on role assignment to grant permissions.

But Postgres and SQL Server rely on their own security - a Postgres role (or Azure SQL user) are labeled to link them with a specific Managed Identity, and all permissions are assigned directly in the database (e.g. `ALTER ROLE` in Postgres or Azure SQL to grant specific permissions in a specific database)

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md (hunk 3 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: I updated this a bit to explain the two paths (basic provisioner with a shared key/bucket and the new Managed identity mode)

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md (hunk 2 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: Fixed.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md (hunk 1 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: Fixed.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md (hunk 4 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-supported-environments.md]: Indeed, updated this to match Azure terminology.

File: content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md (hunk 6 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/secret-store-credentials.md]: Fixed.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 14 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: Indeed, it wasn't correct, I've updated it to reflect what would be stored in the environment's secret.

The provisioner's (storage admin) also has a configuration secret, which contains the SQL Server details above. It's a bit different because e.g. the environment needs to follow the Mendix Runtime format and provide different details (the environment username and database, not the admin username).

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 19 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: Fixed.

File: content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md (hunk 18 above)
Comment (zlogic) [reply content/en/docs/developerportal/deploy/private-cloud/private-cloud-cluster/private-cloud-storage-plans.md]: Fixed - indeed it should have been set to Managed Identity authentication=no

**Comment by zlogic** [comment]
Hi @katarzyna-koltun-mx, we've completed our internal review, could you please take a look when you have the time?

**Comment by katarzyna-koltun-mx** [comment]
Review in progress.

**Review by katarzyna-koltun-mx — APPROVED** [review]

**Comment by katarzyna-koltun-mx** [comment]
Completed review, waiting for the go ahead.
